src/Component/Doctrine/Common/Metadata/Resource/Factory/DoctrineResourceMetadataCollectionFactory.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Doctrine\Common\Metadata\Resource\Factory;
+
+use Sylius\Component\Resource\Doctrine\Common\State\PersistProcessor;
+use Sylius\Component\Resource\Doctrine\Common\State\RemoveProcessor;
+use Sylius\Component\Resource\Metadata\DeleteOperationInterface;
+use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Component\Resource\Metadata\Operations;
+use Sylius\Component\Resource\Metadata\RegistryInterface;
+use Sylius\Component\Resource\Metadata\Resource as ResourceMetadata;
+use Sylius\Component\Resource\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use Sylius\Component\Resource\Metadata\Resource\ResourceMetadataCollection;
+
+class DoctrineResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
```

Comment:
Why didn't you finalized this class? Is there a proxy created from it?
```suggestion
final class DoctrineResourceMetadataCollectionFactory implements ResourceMetadataCollectionFactoryInterface
```

Reply:
Oh no, that's a mistake